src/transformers/modeling_tf_utils.py

```diff
@@ -514,11 +536,17 @@ def load_tf_weights(model, resolved_archive_file, ignore_mismatched_sizes=False,
                 for symbolic_weight in symbolic_weights:
                     # TF names always start with the model name so we ignore it
                     if _prefix is not None:
-                        delimeter = len(_prefix.split("/"))
-                        symbolic_weight_name = "/".join(
-                            symbolic_weight.name.split("/")[:delimeter]
-                            + symbolic_weight.name.split("/")[delimeter + 1 :]
-                        )
+
+                        if symbolic_weight.name in weight_name_map:
+                            # Deal with the case where the model checkpoint is from a model where `load_weight_prefix`
+                            # is already applied.
+                            symbolic_weight_name = weight_name_map[symbolic_weight.name]
+                        else:
+                            delimeter = len(_prefix.split("/"))
+                            symbolic_weight_name = "/".join(
+                                symbolic_weight.name.split("/")[:delimeter]
+                                + symbolic_weight.name.split("/")[delimeter + 1 :]
+                            )
```

Comment:
This part is done to make the following test working.

Basically, that test verifies the encoder/decoder of `TFEncoderDecoderModel` could be saved and loaded individually to form `TFEncoderDecoderModel` using `from_encoder_decoder_pretrained`.

```
  enc_dec_model = TFEncoderDecoderModel(...)
  enc_dec_model.encoder.save_pretrained(...)
  enc_dec_model.decoder.save_pretrained(...)
  enc_dec_model = TFEncoderDecoderModel.from_encoder_decoder_pretrained(
      encoder_pretrained_model_name_or_path=...,
      decoder_pretrained_model_name_or_path=...,
  )
```
https://github.com/huggingface/transformers/blob/f73cab3be8f0dc2cd816ce8f5c9a50e113f8eacb/tests/test_modeling_tf_encoder_decoder.py#L209

Without this change, even for `TFRagModel`, saving then loading individually won't work.

However, it might be arguable that if this `saving then loading individually` really make sense. If not, I will revert this part and remove the corresponding test I added.